.circleci/config.yml

```diff
@@ -1,42 +1,94 @@
 jobs:
-  test:
+  build:
     docker:
-      - image: hysds/pge-base:latest
+      - image: ubuntu:bionic
+        user: root
     steps:
-      - checkout
       - run:
-          name: Test
+          name: Setting Environment Variables
           command: |
-            source $HOME/verdi/bin/activate
-            sudo yum -y update
-            #sudo yum -y install gcc
-            #pip install -U pytest jsonschema
-            mkdir test-reports
-            #pytest --junitxml=test-reports/junit.xml
-
-      - store_test_results:
-          path: test-reports
+            apt-get update
+            apt-get upgrade -y
+            apt-get install -y git
+            apt-get install -y wget
+            # if [ -z ${PYTHONPATH+x} ]; then export PYTHONPATH=""; fi
+            echo $MINTPY_HOME
+            export MINTPY_HOME=~/tools/MintPy
+            export PYTHONPATH=${PYTHONPATH}:${MINTPY_HOME}
+            export PATH=${PATH}:${MINTPY_HOME}/mintpy
+            export PYAPS_HOME=~/tools/PyAPS
+            export PYTHONPATH=${PYTHONPATH}:${PYAPS_HOME}
+      - run:
+          name: Downloading MintPy and PyAPS
+          command: | 
+            export MINTPY_HOME=~/tools/MintPy
+            export PYAPS_HOME=~/tools/PyAPS
+            git clone https://github.com/insarlab/MintPy.git $MINTPY_HOME
+            git clone https://github.com/yunjunz/pyaps3.git $PYAPS_HOME/pyaps3
+            pwd
+            ls
+      - run:
+          name: Installing Dependencies via Conda
+          command: |
+            pwd
+            ls
+            export MINTPY_HOME=~/tools/MintPy
+            export PYTHONPATH=${PYTHONPATH}:${MINTPY_HOME}
+            export PATH=${PATH}:${MINTPY_HOME}/mintpy
+            export PYTHON3DIR=~/tools/miniconda3
+            export PATH=${PATH}:${PYTHON3DIR}/bin
+            export PYAPS_HOME=~/tools/PyAPS
+            export PYTHONPATH=${PYTHONPATH}:${PYAPS_HOME}
+            wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+            chmod +x Miniconda3-latest-Linux-x86_64.sh
+            ./Miniconda3-latest-Linux-x86_64.sh -b -p $PYTHON3DIR
+            $PYTHON3DIR/bin/conda config --add channels conda-forge
+            $PYTHON3DIR/bin/conda install --yes --file $MINTPY_HOME/docs/conda.txt
+            $PYTHON3DIR/bin/pip install git+https://github.com/tylere/pykml.git
+      - run:
+          name: Testing Mintpy on Fernandina
+          command: |
+            export MINTPY_HOME=~/tools/MintPy
+            export PYTHONPATH=${PYTHONPATH}:${MINTPY_HOME}
+            export PATH=${PATH}:${MINTPY_HOME}/mintpy
+            export PYTHON3DIR=~/tools/miniconda3
+            export PATH=${PATH}:${PYTHON3DIR}/bin
+            export PYAPS_HOME=~/tools/PyAPS
+            export PYTHONPATH=${PYTHONPATH}:${PYAPS_HOME}
+            export PATH=${PATH}:${MINTPY_HOME}/test
+            cd ${MINTPY_HOME}/test
+            test_smallbaselineApp.py --dir ${MINTPY_HOME}/test --dset FernandinaSenDT128
```

Comment:
The `--dir` is the test data running directory and we should use a non-code directory here. Can we change somewhere else, e.g. `~/data`?

If we change the following

```
export PATH=${PATH}:${MINTPY_HOME}/test
cd ${MINTPY_HOME}/test
test_smallbaselineApp.py --dir ${MINTPY_HOME}/test --dset FernandinaSenDT128
```

to

```
${MINTPY_HOME}/test/test_smallbaselineApp.py --dir ~/data --dset FernandinaSenDT128
```

and it still works, that would be preferred.

Reply:
Same for the other two datasets.

Reply:
@yunjunz would definitely be possible, you can also control it in circle ci and set them as top-level environment variables. Same with shiedling credentials if you ever want to test some of the APS related workflows.

Reply:
Thank you @dbekaert, testing PyAPS sounds great, I will add it to the to-do-list.

Reply:
CircleCI has a solution to save env. var. in different level (Project, Container, ...), but it didn't work as I expected. Env. var couldn't be retrieved. 

just for testing your codes it would be OK. I try to solve it later